repology/parsers/parsers/cygwin.py

```diff
@@ -0,0 +1,63 @@
+# Copyright (C) 2019 Jon Turney <jon.turney@dronecode.org.uk>
+#
+# This file is part of repology
+#
+# repology is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# repology is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with repology.  If not, see <http://www.gnu.org/licenses/>.
+
+from typing import Generator
+
+from jsonslicer import JsonSlicer
+
+from repology.package import PackageFlags
+from repology.packagemaker import PackageFactory, PackageMaker
+from repology.parsers import Parser
+from repology.transformer import PackageTransformer
+
+
+class CygwinParser(Parser):
+    def iter_parse(self, path: str, factory: PackageFactory, transformer: PackageTransformer) -> Generator[PackageMaker, None, None]:
+        with open(path, 'rb') as jsonfile:
+            for packagedata in JsonSlicer(jsonfile, ('packages', None)):
+                # packages with names starting with an underscore are
+                # uninteresting as they contain cygwin-specific
+                # installation helper scripts
+                if packagedata['name'].startswith('_'):
+                    continue
+
+                if '_obsolete' in packagedata['categories']:
+                    continue
+
+                for maturity in ['stable', 'test']:
+                    if maturity not in packagedata['versions']:
+                        continue
+
+                    version = packagedata['versions'][maturity][-1]
+
+                    for subpackage in packagedata['subpackages']:
+                        pkg = factory.begin()
+
+                        pkg.set_basename(packagedata['name'])
```

Comment:
Sorry, perhaps I wasn't too clear. I haven't meant generic fields like basename, categories, summary and maintainers to be set in the inner loop. They belong outside of the loop, and this makes logs produced by PackageMakager more useful as basename can be used to identify the package which produces output. In the inner loop, just `clone` pkg and set subpackage specific fields like version, flags and name.

Reply:
Ah, right.  I think I understand what you meant now.